### PR TITLE
[refactor]: bump rust version to 1.58.

### PR DESCRIPTION
**Pull request by appetrosyan**
Signed-off-by: Aleksandr <a-p-petrosyan@yandex.ru>

### Description of the Change

Bumped the version of `rust` and satisfied the lints added in 1.58. 

### Issue

None

### Benefits

Update

### Possible Drawbacks

None

### Usage Examples or Tests *[optional]*

`rustup toolchain update stable && cargo lints clippy --workspace --benches --tests --examples`

### CODE REVIEW

**Comment by appetrosyan** [comment]
Updating the CI container at the moment. Should fix the static analysis.

File: crypto/src/lib.rs [hunk 1]
```diff
@@ -133,18 +133,21 @@ pub struct KeyGenConfiguration {
 
 impl KeyGenConfiguration {
     /// Use seed
+    #[must_use]
```
Comment (mversic) [inline crypto/src/lib.rs:136]: You can also mark structures as `must_use` if it makes sense. I think an example is `Result`

Comment (appetrosyan) [reply crypto/src/lib.rs:136]: And traits. I'm trying not to mark anything that wasn't suggested by the new lints.

**Comment by e-ivkov** [comment]
We faced the following issue, preventing from merging this PR for now https://github.com/rust-lang/rust-clippy/issues/8307

**Comment by appetrosyan** [comment]
The issue fix is not merged into rust 1.58.1. Thus it may well be that we will need to skip this version altogether, and wait until 1.59.

**Pull request closed without merging**